File: AUGUST 24/23 August/Left View of Binary Tree gfg.cpp
void traverse(Node* root, vector<int> &view, int depth)
{
    if(root==nullptr)
        return;
    if(depth>view.size())
        view.push_back(root->data);
    traverse(root->left, view, depth+1);
    traverse(root->right, view, depth+1);
}
vector<int> leftView(Node *root)
{
   // Your code here
   vector<int> view;
   traverse(root, view, 1);
   return view;
}

File: MAY 2025/06 May/Left View of Binary Tree gfg.java
class Solution {
    // Function to return list containing elements of left view of binary tree.
    
    ArrayList<Integer> leftView(Node root) {
        // code here
        ArrayList<Integer> list = new ArrayList<>();
        Queue<Node> q = new LinkedList<>();
        if(root != null)
            q.add(root);
        
        while(!q.isEmpty())
        {
            list.add(q.peek().data);
            int n = q.size();
            for(int i=0; i<n; i++)
            {
                Node node = q.poll();
                if(node.left != null)
                    q.add(node.left);
                if(node.right != null)
                    q.add(node.right);
            }
        }
        return list;
    }
}
